### PR TITLE
Handle string literals for constant expression evaluator

### PR DESCRIPTION
**Pull request by blitz-1306**
## Changes
- [x] Add support for string literals for equality and comparison ops in constant expression evaluator.

An example:
```solidity
contract Sample {
    function test() public {
        assert(bytes1(0x58) == 'X');
        assert(bytes2(0x5859) == 'XY');
        assert(bytes1(0x58) == hex'58');
        assert(bytes2(0x5859) == hex'5859');
        assert(bytes1(0x00) == '');
        assert(bytes1(0x00) == hex'');
    }
}
```

Regards.

### CODE REVIEW

File: src/types/eval_const.ts [hunk 1]
```diff
@@ -220,19 +228,16 @@ export function evalBinaryImpl(operator: string, left: Value, right: Value): Val
...
             throw new EvalError(
                 `${operator} not allowed for strings ${str(left)} and ${str(right)}`
             );
         }
 
-        let isEqual: boolean;
+        const leftDec = promoteToDec(left);
+        const rightDec = promoteToDec(right);
 
-        if (left instanceof Decimal && right instanceof Decimal) {
```
Comment (cd1m0) [inline src/types/eval_const.ts:231]: I think this change may break constant eval of `true == true`, since bool is not promotable to Decimal.

Comment (cd1m0) [reply src/types/eval_const.ts:231]: Maybe change this to:

```
if (left instanceof Decimal || right instanceof Decimal) {
        const leftDec = promoteToDec(left);
        const rightDec = promoteToDec(right);
        
        isEqual = leftDec.equals(rightDec);
} else {
       isEqual = left === right;
}

**Review by cd1m0 — CHANGES_REQUESTED** [review]
One nit

File: src/types/eval_const.ts (hunk 1 above)
Comment (blitz-1306) [reply src/types/eval_const.ts:231]: Good note, will fix.

Comment (blitz-1306) [reply src/types/eval_const.ts:231]: Addressed slightly different way in d06e427. Consider.

**Review by cd1m0 — APPROVED** [review]

**Pull request merged**